### PR TITLE
Root

### PR DESCRIPTION
**Pull request by arb**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,3 +3,4 @@ language: node_js
 node_js:
   - "4"
   - "5"
```
Comment (cjihrig) [inline .travis.yml]: Version 5 can be dropped. It is EOL.

File: README.md [hunk 2]
```diff
@@ -7,14 +7,15 @@
...
-## `new WhiteOut (filter, [options])`
+## `new WhiteOut (filter, [root], [options])`
 
 Creates a new `WhiteOut` transform stream with the following arguments.
 - `filter` - a key value pair where `key` is the property value to censor on the object payload and the `value` is the censor type.
   - Valid options for `value` are "censor", "remove" and a RegExp.
     - "censor" - replaces the value with a string of "X"s.
     - "remove" - completely removes the key from object
     - Anything else will be treated as a `RegExp` definition and will be passed into `new RegExp`.
+- `[root]` - an object string path (ex `'response.payload'`) that will be used when the censor algorithm starts. Useful for censoring only a subsection of the entire `data` object. Defaults to `undefined` which means the entire `data` object will be traversed. For performance reasons, it is recodmended to set `root` to only the specific segment of `data` you wish to filter.
```
Comment (cjihrig) [inline README.md]: Why are there brackets around root?

File: index.js [hunk 3]
```diff
@@ -13,7 +15,8 @@ const replacer = (match, group) => {
 };
 
 class WhiteOut extends Stream.Transform {
-  constructor (rules, options) {
+  constructor (rules, root, options) {
```
Comment (cjihrig) [inline index.js]: Seems strange to keep adding arguments to the constructor when there is already an `options` object.

File: index.js [hunk 4]
```diff
@@ -13,7 +15,8 @@ const replacer = (match, group) => {
...
+    Assert.ok(root == null || typeof root === 'string', 'root must be a string'); // eslint-disable-line eqeqeq
```
Comment (cjihrig) [inline index.js]: Why allow `null` here? Why not just permit `undefined`.

File: README.md (hunk 2 above)
Comment (arb) [inline README.md]: Because it's optional.

File: index.js (hunk 3 above)
Comment (arb) [inline index.js]: Those are options for the base stream constructor.

File: README.md (hunk 2 above)
Comment (cjihrig) [inline README.md]: Yea, you're only supposed to do that in the function signature though.

File: index.js (hunk 4 above)
Comment (arb) [inline index.js]: I was thinking you'd pass `null` if you want to set `options` and not `root`. If you want to set something to "not set" I usually go with `null`, not setting to `undefined`.

Comment (cjihrig) [inline index.js]: I guess that makes sense. Although it would make more sense to only support a single `options` object.

File: README.md (hunk 2 above)
Comment (arb) [inline README.md]: I've done it in a bunch of places... in our org and the hapi org as well. Not married to it and if you want me to change, I will.

Comment (cjihrig) [inline README.md]: Hm, I never noticed before. Just pointing it out. Feel free to address it or not.

File: index.js [hunk 5]
```diff
@@ -1,7 +1,9 @@
 'use strict';
 
+const Assert = require('assert');
 const Stream = require('stream');
 const Traverse = require('traverse');
+const Reach = require('reach');
```
Comment (cjihrig) [inline index.js]: Reach should come before Traverse.

File: index.js [hunk 6]
```diff
@@ -14,7 +16,10 @@ const replacer = (match, group) => {
 
 class WhiteOut extends Stream.Transform {
   constructor (rules, options) {
-    super(Object.assign({}, options, { objectMode: true }));
+    options = options || {};
+    Assert.ok(options.root == null || typeof options.root === 'string', 'root must be a string'); // eslint-disable-line eqeqeq
```
Comment (cjihrig) [inline index.js]: Now this can check for `undefined`.

File: index.js [hunk 7]
```diff
@@ -14,7 +16,10 @@ const replacer = (match, group) => {
 
 class WhiteOut extends Stream.Transform {
   constructor (rules, options) {
-    super(Object.assign({}, options, { objectMode: true }));
+    options = options || {};
```
Comment (cjihrig) [inline index.js]: This line and the `Object.assign()` could probably be combined if you add an assertion on `objectMode` being `true`. Up to you though.

**Pull request merged**